tools/changelog/changelog/entry.go

```diff
@@ -0,0 +1,60 @@
+package changelog
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Changelog struct {
+	// One or more changes
+	Changes []*Change `yaml:"changes,omitempty"`
+}
+
+func (cl *Changelog) Merge(other Changelog) Changelog {
+	var changes []*Change
+	changes = append(changes, cl.Changes...)
+	changes = append(changes, other.Changes...)
+	return Changelog{
+		Changes: changes,
+	}
+}
+
+// Change records a change by its category, the scope of the change, and metadata from GitHub needed to render a
+// changelog.
+//
+// Follows the conventional commits specification: https://www.conventionalcommits.org/en/v1.0.0/#specification
+type Change struct {
+	// Type is a noun describing the category of change: improvement, fix, and so on.
+	Type string `yaml:"type,omitempty"`
+	// Scope is a noun describing the part of the system improved, such as: cli, codegen, sdk, automation. Subscopes can
+	// be represented with a slash after a scope, as in: cli/about, cli/display, auto/go, codegen/dotnet, sdk/nodejs.
+	Scope Scope `yaml:"scope,omitempty"`
+	// Description is a description of the change.
+	Description string `yaml:"description,omitempty"`
+	// Metadata contains additional, optional data for rendering the changelog, which may be set to override inferred
+	// values.
+	GitHubMeta GitHubMeta `yaml:"github,inline"`
+}
+
+type GitHubMeta struct {
+	// PullRequestNumbers are the GitHub Pull Requests used to implementing this change, typically a single PR.
+	PullRequestNumbers []int `yaml:"prs,omitempty"`
+}
+
+func (v *Change) Conventional() string {
+	var message string
+
+	typ := v.Type
+	message += fmt.Sprintf("%v(%v): %v.", typ, v.Scope.String(), v.Description)
+
+	var prs []string
```

Comment:
```suggestion
	var typ = v.Type
	var message = fmt.Sprintf("%v(%v): %v.", typ, v.Scope.String(), v.Description)
	var prs []string
```
Might be worth inlining the `typ` variable too since it's only used once.